### PR TITLE
Refactor vision blackout procedure

### PR DESCRIPTION
**Pull request by g3force**

### CODE REVIEW

File: 2022/chapters/procedure.adoc [hunk 1]
```diff
@@ -1,150 +1,116 @@
...
-will be given, but the robot should not move during the stop command. This is
-just to allow teams a few seconds to initialize any internal state. The run will
-officially start when a force start command is sent. This will start the timer
-, and at this point, the robot may begin moving.
+There are multiple stages with increasing difficulty.
+Participants will have 3 trials for each stage.
+All trials will be considered for the overall score.
 
-=== Stage 1: Grab a Stationary Ball
+Each trial will be attempted by all participants, who after another.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
Each trial will be attempted by all participants, one after another.
```

File: 2022/chapters/procedure.adoc [hunk 2]
```diff
@@ -1,150 +1,116 @@
...
+The starting configuration of the trial will be similar for each team to achieve same conditions for all participants.
+
+If a participant can not attempt a trial for what ever reasons, no points are awarded for this trial.
+Other trials are not effected.
+
+The robots may move anywhere on the field within the walls.
+The ball may similarly move anywhere on the field throughout the time.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
Similarly, the ball may move anywhere on the field throughout the time.
```

File: 2022/chapters/procedure.adoc [hunk 3]
```diff
@@ -1,150 +1,116 @@
...
+=== Referee commands
 
-For each trial, a random robot starting location/orientation and a
-random ball starting position will be chosen. Both the robot and the
-ball will start stationary. There are no restrictions on the starting
-configuration other than that both robot and ball will be somewhere on
-the field.
+Each trial will start with a *HALT* command.
+Just before the start, a *STOP* command will be given to allow participants to prepare.
+When a *FORCE START* command is sent the timer is started and robots may move.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
When a *FORCE START* command is sent, the timer is started and robots may move.
```

File: 2022/chapters/procedure.adoc [hunk 4]
```diff
@@ -1,150 +1,116 @@
...
+The trial will be stopped with a *HALT* command, when it was successful or time run out.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
The trial will be stopped with a *HALT* command after successful challenge completion or after the time ran out
```

File: 2022/chapters/procedure.adoc [hunk 5]
```diff
@@ -1,150 +1,116 @@
...
-Before each run, the robot will begin in the halt state. A stop command
-will be issued to ready the robot, but the robot should not yet
-move. The run starts with a force start command, at which point the
-timer starts, and the robot may attempt to complete the task.
+=== Scoring
 
-The robot may move anywhere on the field within the walls. The ball
-may similarly move anywhere on the field throughout the time. Each
-trial will last a maximum of one minute.
+Stage may define a set of criteria that give positive or negative points.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
Each stage may define a set of criteria which give positive or negative points.
```

File: 2022/chapters/procedure.adoc [hunk 6]
```diff
@@ -1,150 +1,116 @@
...
+Points will also be given if the overall objective was not achieved.
+The duration of the trial is used as a tie-breaker (faster is better).
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
The duration of a trial is used as a tie-breaker (faster is better).
```

File: 2022/chapters/procedure.adoc [hunk 7]
```diff
@@ -1,150 +1,116 @@
...
+
+=== Stage 1: Grab a Stationary Ball
+
+This stage is designed to be the easiest and demonstrate basic sensing and control capabilities.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
This stage is designed to be the easiest and to demonstrate basic sensing and control capabilities.
```

File: 2022/chapters/procedure.adoc [hunk 8]
```diff
@@ -1,150 +1,116 @@
...
+The robot will need to detect a ball on the field, navigate to it, and place it on its dribbler.
+
+==== Procedure
+
+- The supervisor defines a random robot starting position and orientation
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion

```

File: 2022/chapters/procedure.adoc [hunk 9]
```diff
@@ -1,150 +1,116 @@
...
+- The supervisor defines a random ball starting position
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
- The supervisor defines random starting positions for robot and ball, as well as a random orientation for the robot.
```

File: 2022/chapters/procedure.adoc [hunk 10]
```diff
@@ -1,150 +1,116 @@
...
-zone. Unlike regular gameplay, the ball does not need to stay on the
-ground in order for the goal to be counted.
-
-Before each run the robot will begin in the halt state. A stop command
-will be issued to ready the robot, but the robot should not yet
-move. The run begins with a force start command, at which point the
-timer starts and the robot may attempt to complete the task. Each
-trial will last a maximum of one minute. The trial will stop when the
-goal has occurred or time has elapsed.
+This stage requires that robots can localize a goal and itself The robot will need to detect the ball, move to it and score a goal.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
This stage requires robots to localize themselves as well as the ball on the playing field, move to the ball and score a goal.
```

File: 2022/chapters/procedure.adoc [hunk 11]
```diff
@@ -1,150 +1,116 @@
...
-
-Robots have 1 minute to reach the defined location. The trial
-will end when the robot has stopped for 10 seconds (after having moved
-at least once since the trial start) or the time has elapsed. The 10
-seconds stoppage will be subtracted from the total time used to
-determine tie-breaking.
+==== Procedure
+
+- The supervisor defines a random robot target position
+- The supervisor defines a random robot starting position seed
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
- The supervisor defines a random target position, random robot orientation and a random start position seed.
```

File: 2022/chapters/procedure.adoc [hunk 12]
```diff
@@ -1,150 +1,116 @@
...
+- The trial is started
+- The trial is stopped after one minute or if the robot did not move for 10 seconds after moving at least once
 
 ==== Scoring
 
-Score is equal to the negative straight line distance between the
-robot's final position and the destination position. In other words,
-closer is better.
+Score is equal to the negative straight line distance between the robot's final position and the destination position.
+In other words, closer is better.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
The score is equal to the negative straight line distance between the robot's final position and the destination position.
(In other words, closer is better)
```

File: 2022/chapters/procedure.adoc [hunk 13]
```diff
@@ -1,150 +1,116 @@
...
+
+- The supervisor defines a random robot starting position and orientation
+- The supervisor defines a random ball starting position
+- Robot and ball are moved to their starting positions
+- The trial is started
+- The trial is stopped after one minute or if the robot scored a goal
+
+==== Additional Rules
+
+- Either of the two goals can be used for scoring
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
- Either of the two goals may be used for scoring
```

File: 2022/chapters/procedure.adoc [hunk 14]
```diff
@@ -1,150 +1,116 @@
...
+Score is equal to the negative straight line distance between the robot's final position and the destination position.
+In other words, closer is better.
 
 === Stage 4: Scoring an Indirect Goal (without Obstacles)
 
-This stage is a more complex version of stage 2 and 3. The objective is the same,
-but now robot must pass the ball to a second robot. Different from stage 2, the
-robots should score a goal on the positive side of the field.
+The objective of this stage is to score an indirect goal using two robots.
+Robots are required to recognize other robots, receive a ball and coordinate between each other.
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
Robots are required to recognize other robots, receive a ball, coordinate between each other and finally to score a goal.
```

File: 2022/chapters/procedure.adoc [hunk 15]
```diff
@@ -1,150 +1,116 @@
...
 
-One robot will start at a position (0.5m, 1.5m) and another in (0.5m, -1.5m), both
-orientations will be at 0. The ball position will start at the rambom position in the 
-positive side of the field. To maintain the difficulty, the robot distance from
-one of the robots will be equal between different teams. While scoring, the robots 
-may not enter the defense zone.
+- The supervisor defines a random ball starting position seed in the positive field half
+- Robot 1 is placed at (0.5 m, 1.5 m, 0°)
+- Robot 2 is placed at (0.5 m, 1.5 m, 0°)
+- The ball is placed at a random ball starting position that is within 0.5 m distance to the position seed
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
- The ball is placed at a random starting position within 0.5 m distance of the position seed.
```

File: 2022/chapters/procedure.adoc [hunk 16]
```diff
@@ -154,4 +120,4 @@ may not enter the defense zone.
 - Pass hits the receiver robot dribbler -- +1
 - Ball entered defense zone -- +1
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
 - Ball entered a defense zone -- +1
```

File: 2022/chapters/procedure.adoc [hunk 17]
```diff
@@ -154,4 +120,4 @@ may not enter the defense zone.
...
 - Ball entered goal -- +1
-- Robot entered defense zone -- -1
\ No newline at end of file
+- Robot entered defense zone -- -1
```
Comment (AlexSchmitz222) [inline 2022/chapters/procedure.adoc]:
```suggestion
- Robot entered a defense zone -- -1
```

**Review by AlexSchmitz222 — COMMENTED** [review]
I fixed a few small typos and made some propositions for shortening the challenge rules.

File: 2022/chapters/procedure.adoc [hunk 18]
```diff
@@ -1,157 +1,121 @@
...
+This stage is designed to demonstrate localization capabilities without the use of the global vision system.
+No ball will be used for this stage.
+
+==== Procedure
+
+- The supervisor defines a random target position, random robot orientation and a random start position seed.
+- Participants may enter the staring position seed and the target position into their robot configuration
+- The supervisor defines a random robot starting orientation
+- The supervisor defines a random robot starting position that is within 0.5 m distance to the position seed
+- the Robot is moved to its starting positions and orientation
```
Comment (lhcavalcanti) [inline 2022/chapters/procedure.adoc]:
```suggestion
- The robot is moved to its starting positions and orientation
```

File: 2022/chapters/procedure.adoc [hunk 19]
```diff
@@ -1,157 +1,121 @@
...
-zone. Unlike regular gameplay, the ball does not need to stay on the
-ground in order for the goal to be counted.
-
-Before each run the robot will begin in the halt state. A stop command
-will be issued to ready the robot, but the robot should not yet
-move. The run begins with a force start command, at which point the
-timer starts and the robot may attempt to complete the task. Each
-trial will last a maximum of one minute. The trial will stop when the
-goal has occurred or time has elapsed.
+This stage requires robots to localize themselves as well as the ball on the playing field, move to the ball and score a goal.
```
Comment (lhcavalcanti) [inline 2022/chapters/procedure.adoc]: I don't think robots needs to localize themselves here, don't the ball and the goal are sufficient?

**Review by lhcavalcanti — APPROVED** [review]
Looks really good to me, made just two small comments. Thanks for that help Nicolai

File: 2022/chapters/procedure.adoc (hunk 18 above)
Comment (AlexSchmitz222) [reply 2022/chapters/procedure.adoc]: oops, my mistake :D

also: 
```suggestion
- The robot is moved to its starting position and orientation.
```

File: 2022/chapters/procedure.adoc (hunk 19 above)
Comment (g3force) [reply 2022/chapters/procedure.adoc]: true, changed it from themselves to goal

**Pull request merged**